website/project/tasks.py

```diff
@@ -44,8 +44,7 @@ def update_collecting_metadata(node, saved_fields):
         if node.is_public:
             update_collected_metadata(node._id)
         else:
-            if 'is_public' in saved_fields:
-                update_collected_metadata(node._id, op='delete')
+            update_collected_metadata(node._id, op='delete')
```

Comment:
Minor: May lead to ES 404's, but that's better than private nodes in search.

Reply:
https://github.com/CenterForOpenScience/osf.io/pull/9358/files#diff-869967c44a8c8a275d2043563fef5158R30 will probably keep this from being hit unnecessarily.